File: JavaScript/Two Pointer/11. Container With Most Water.js
/* 11. Container With Most Water
Difficulty: Medium
Submission Link: https://leetcode.com/problems/container-with-most-water/submissions/1247602250/ */

var maxArea = function(height) {
    let left = 0;
    let right = height.length - 1;
    let maxWater = 0;
    while (left < right) {
        let width = right - left;
        let h = Math.min(height[left], height[right]);
        let water = width * h;
        maxWater = Math.max(maxWater, water);
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    return maxWater;
};

File: C#/Two Pointer/11. Container With Most Water.cs
/* 11. Container With Most Water
Difficulty: Medium
Submission Link: https://leetcode.com/problems/container-with-most-water/submissions/1247602868/ */

public class Solution {
    public int MaxArea(int[] height) {
        int maxArea = 0;
        int left = 0, right = height.Length - 1;

        while (left < right) {
            int h = Math.Min(height[left], height[right]);
            int width = right - left;
            maxArea = Math.Max(maxArea,h*width);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxArea;        
    }
}